src/mxbase.jl

```diff
@@ -42,24 +42,24 @@ function get_paths()
         end
         default_startcmd = "exec $(Base.shell_escape(default_startcmd)) -nosplash"
     elseif is_windows()
-        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.WORD_SIZE == 32 ? "win32" : "win64"), "MATLAB.exe")
+        default_startcmd = joinpath(matlab_homepath, "bin", (Sys.ARCH === :i686  ? "win32" : "win64"), "MATLAB.exe")
         if !isfile(default_startcmd)
             error("The MATLAB path is invalid. Set the MATLAB_HOME evironmental variable to the MATLAB root.")
         end
         default_startcmd *= " -nosplash"
     end
 
     # Get path to MATLAB libraries
-    if is_linux()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "glnx86" : "glnxa64"
+    matlab_library_dir = if is_linux()
+        Sys.ARCH === :i686 ? "glnx86" : "glnxa64"
     elseif is_apple()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "maci" : "maci64"
+        Sys.ARCH === :i686 ? "maci" : "maci64"
     elseif is_windows()
-        matlab_library_dir = Sys.WORD_SIZE == 32 ? "win32" : "win64"
+        Sys.ARCH === :i686 ? "win32" : "win64"
```

Comment:
In any case, `It's odd but running Sys.ARCH on a 64-bit machine using 32 bit julia, spits out x86_64 instead of i686, isin't this a bug?` sounds wrong but your output suggests that it isn't happening.

Also, I think this can be closed since there's no reason to do this.